backend/common/utils/concurrency.py

```
@@ -77,10 +77,89 @@ def _env_int(name: str, default: int) -> int:
+        """
+        Release a semaphore slot and update the counter atomically.
+        
+        Note: This is a sync method. We update the counter by scheduling
+        an async task. If no event loop is available, we use a best-effort
+        synchronous update.
+        """
+        self._semaphore.release()
+        # Try to update counter via async task (preferred)
+        try:
+            loop = asyncio.get_running_loop()
+            loop.create_task(self._increment_available())
```

Ruff (0.14.8)

Store a reference to the return value of `loop.create_task`

(RUF006)